### PR TITLE
Fix/html validation forms 

### LINKED ISSUE #36270: Fix User name and other parameters potentially containing HTLM characters
### Body

The User name and other parameters should be  validated to not contain characters that might contain HTML-invalid characters. Also when they do, they should be escaped when rendered on HTML pages.

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by chrfoyer**
closes: #36270 

Validate the text provided when creating a connection to avoid html tags. This is done by creating a new validator and using it for said text field. Test were run and pre commit was run, causing a variety of other binaries to be recreated

### CODE REVIEW

**Comment by potiuk** [comment]
I think you have quite a lot of conflicts because you are 213 commits behind. I suggest to rebase, resolve conflicts (or rather reapply your changes staarting from main):

Screenshot 2024-02-10 at 22 11 41

**Comment by chrfoyer** [comment]
> I think you have quite a lot of conflicts because you are 213 commits behind. I suggest to rebase, resolve conflicts (or rather reapply your changes staarting from main):
> 
> Screenshot 2024-02-10 at 22 11 41
> 

Thanks for your feedback! I will do that and try again

File: airflow/www/forms.py [hunk 1]
```diff
@@ -234,13 +234,42 @@ def process(self, formdata=None, obj=None, **kwargs):
...
+            lazy_gettext("Password"),
+            widget=BS3PasswordFieldWidget(),
+            validators=[NoHTMLTags()],
+        )
+        port = IntegerField(
+            lazy_gettext("Port"),
+            validators=[Optional()],
+            widget=BS3TextFieldWidget(),
+        )
+        extra = TextAreaField(
```
Comment (hussein-awala) [inline airflow/www/forms.py]: Just checked the issue and found:
>  Also when they do, they should be escaped when rendered on HTML pages.

Also, I'm wondering if some users need to pass HTML via extra; should we block that completely or just stop rendering the HTML by escaping them on the connection web UI?

Comment (chrfoyer) [reply airflow/www/forms.py]: I think escaping is probably the better option. I will get on that

**Comment by chrfoyer** [comment]
I am changing my approach as well as getting the more up to date version of main. I will link to the new PR when ready

**Pull request closed without merging**